docs/ruleset_functions/get_spaces_served_by_SWH_use.md

```diff
@@ -0,0 +1,39 @@
+## get_spaces_served_by_SWH_use
+
+Description: This function determines the spaces served by a given SWH use.  The rule is that if spaces reference the swh_use, then the use applies to only those spaces.  If the use references no spaces, it applies to all spaces in the building segment.
+
+
+Inputs:
+- **RMD**
+- **swh_use**
+
+Returns:
+- **spaces_served**: a list of space ids
+
+Function Call:
+
+
+Data Lookup: None
+
+Logic:
+
+- find the building_segment that contains this swh_use by looking through each building segment: `for building_segment in RMD.building_segments:`
+    - check if the swh_use is in this building segment: `if swh_use in building_segment.service_water_heating_uses:`
+        - create a list of spaces: `spaces = []`
+        - look at each building segment: `for building_segment in RMD.building_segments`
+            - check if the swh_use is in this building_segment: `if swh_use in building_segment.service_water_heating_uses:`
+                - look at each space in the building segment to see which of them reference the swh_use: `for space in building_segment.spaces:`
+                    - if the swh_use ID is in the list of space ServiceWaterHeatingUses, then the space is one of the spaces and we add it to the list: `if swh_use.id in space.service_water_heating_uses: spaces << space`
+        - here, we check whether there are any spaces applied to the swh_use.  If no spaces are applied, and the service water heating use units are not POWER or VOLUME, then this use applies to all spaces in the building_segment: `if len(spaces) == 0:`
+            - check if the swh use use_units are not POWER or VOLUME.  Service water heating uses with use units power or volume are only counted once, unless they are expicitly assigned to multiple spaces: `if swh_use.use_units not in ["POWER","VOLUME"]: spaces = building_segment.spaces`
+        - convert the list of spaces to a list of space ids: `space_ids = [space.id for space in spaces]
```

Comment:
we return a list of IDs here, but the first place I am seeing this function called then uses the get_obj_by_id() to go and get the object again. Consider returning the list of objects.

Reply:
I thought the convention for the RCT was to always pass ids and not objects? I’d be happy to pass objects instead! Can you confirm?

Reply:
My thought on this is that it is something the RCT team would handle during implementation and that we did not have to worry too much about it as long as the intent was communicated.